### ISSUE TITLE
AlertDialog.actions overrides ButtonTheme 

### ISSUE THREAD

**Issue by Levi-Lesches** [flutter/flutter#38646]
When making my app, I noticed the buttons in `AlertDialog.actions` looked a bit off -- they weren't conforming to `MaterialApp.theme.buttonTheme.buttonColor`. After checking the documentation, I found this under `AlertDialog.build`: 

```dart
if (actions != null) {
  children.add(ButtonTheme.bar(
    child: ButtonBar(
      children: actions,
    ),
  ));
}
```

Why does AlertDialog provide its own `ButtonTheme` if it's not even going to give it an actual theme? 

A temporary workaround I found was using the ambient `ButtonTheme.colorScheme`. So, if you want to make, say, a `RaisedButton` with the correct color, you could do: 

```dart
final MaterialButton _button = RaisedButton(onPressed: () {});

// the build method

actions: [
  RaisedButton(
    child: Text ("OK"),
    onPressed: () {},
    color: Theme.of(context).buttonTheme.getFillColor(_button),
  )
]
// more code below
```

Output of `flutter doctor -v`: 
```
[√] Flutter (Channel stable, v1.7.8+hotfix.4, on Microsoft Windows [Version
    10.0.18362.239], locale en-US)
    • Flutter version 1.7.8+hotfix.4 at C:\Users\levi\flutter
    • Framework revision 20e59316b8 (4 weeks ago), 2019-07-18 20:04:33 -0700
    • Engine revision fee001c93f
    • Dart version 2.4.0


[√] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at C:\Users\levi\AppData\Local\Android\sdk
    • Android NDK location not configured (optional; useful for native profiling
      support)
    • Platform android-28, build-tools 28.0.3
    • Java binary at: C:\Program Files\Android\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b01)
    • All Android licenses accepted.

[√] Android Studio (version 3.4)
    • Android Studio at C:\Program Files\Android
    • Flutter plugin version 35.3.1
    • Dart plugin version 183.6270
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b01)

[√] Connected device (1 available)
    • SM G950U1 • 988861314d554d4f52 • android-arm64 • Android 9 (API 28)

• No issues found!
```

**Comment by Levi-Lesches** [comment]
@darrenaustin I stumbled across your PR #37544. Does that fix this issue?

**Comment by Levi-Lesches** [comment]
Sorry I still haven't had time to fully test this. I'm trying to rush a tight deadline, but I found something interesting here. I was making a companion app for my original app (an admin console which required access that normal users should not have/need), and while trying to make a dialog I noticed that the RaisedButton was the default color of blue (I hadn't overridden the theme yet), but so was the text! I had a FlatButton next to it and its text color was also that default blue, so I manually changed the RaisedButton's child text color to white, which solved the problem. Here is the code for that: 
```dart
void showError(BuildContext context) => showDialog(
  context: context,
  builder: (_) => AlertDialog(
    title: Text ("Login failed"),
    content: Column(
      mainAxisSize: MainAxisSize.min,
      children: [
        Text (
          "Could not log into your account",
          textScaleFactor: 1.5,
        ),
        SizedBox(height: 10),
        Text(
          "A problem prevented you from logging in. "
          "Please contact someone from IT for help."
        )
      ],
    ),
    actions: [
      RaisedButton(
        child: Text (
          "Close",
          style: TextStyle(color: Colors.white),
        ),
        onPressed: Navigator.of(context).pop,
      )
    ]
  )
);
```
(Note that I later got rid of the FlatButton)

**Comment by Levi-Lesches** [comment]
Ok so I properly started a new project yadda yadda. Basically this code: 
```dart
import "package:flutter/material.dart";

const Color mySpecificBlue = Color(0xFF004B8D);

void main() => runApp(MainApp());

class MainApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) => MaterialApp(
    home: DialogPage(),
  );
}

class DialogPage extends StatelessWidget {
  @override
  Widget build(BuildContext context) => Scaffold(
    floatingActionButton: FloatingActionButton(
      child: Icon(Icons.info),
      onPressed: () => showDialog(
        context: context,
        builder: (_) => AlertDialog(
          title: Text ("This is a test"),
          actions: [
            RaisedButton(
              child: Text ("OK"),
              onPressed: () => Navigator.of(context).pop()
            )
          ]
        )
      )
    )
  );
}

```
yields this: 
![flutter_01](https://user-images.githubusercontent.com/20747538/64084129-1f42bf80-ccf6-11e9-9882-348fe5c4a874.png)
Which is new, since now I can't even see the text. 

Giving the theme a new `buttonColor` and `buttonThemeData.buttonColor` (which one is preferrable? The docs should be a bit more explicit on this...) like so yields the same result: 
```dart
class MainApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) => MaterialApp(
    home: DialogPage(),
    theme: ThemeData(
      buttonColor: mySpecificBlue,
      buttonTheme: ButtonThemeData(
        buttonColor: mySpecificBlue,
      )
    )
  );
}
```
The only thing that seemed to work was giving `buttonTheme` a `ColorScheme`: 
```dart
class MainApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) => MaterialApp(
    home: DialogPage(),
    theme: ThemeData(
      buttonTheme: ButtonThemeData(
        colorScheme: ColorScheme.light(
          primary: mySpecificBlue,
          onPrimary: Colors.white,
        ),
      )
    )
  );
}
```
which gives this: 
![flutter_02](https://user-images.githubusercontent.com/20747538/64084305-266acd00-ccf8-11e9-8f54-9a33c1840317.png)

This reveals that the text color was indeed the default blue all along. Additionally, `onPrimary` was not used for the text color. To get the text color to change I had to provide a `secondary` to `ColorScheme`: 
```dart
class MainApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) => MaterialApp(
    home: DialogPage(),
    theme: ThemeData(
      buttonTheme: ButtonThemeData(
        colorScheme: ColorScheme.light(
          primary: mySpecificBlue,
          secondary: Colors.white,
        ),
      )
    )
  );
}
```
Which fixes it: 
![flutter_03](https://user-images.githubusercontent.com/20747538/64084377-efe18200-ccf8-11e9-9c74-e5ff75ad3a36.png)

This is also discussed at issue #22745, but nobody seems to be looking into it. If you look at my comments there one could think to set the text color to `primary` or `secondary`, but as @willlarche points out, Material design dictates careful consideration here.

**Comment by Levi-Lesches** [comment]
So to summarize, the problems are: 

1. The button's text color defaults to the primary color, which is the same as the button's default fill color.
2. Adding `buttonColor` does not change the color in any way, only `ThemeData.buttonTheme.colorScheme.primary` affects the color.
3. Adding `onPrimary` does not change the text color. Only `ColorScheme.secondary` is used for text color.

**Comment by Levi-Lesches** [comment]
Any update on this? I would say it's a pretty big issue seeing as none of this is documented right now.

**Comment by Levi-Lesches** [comment]
Hey @darrenaustin, I found why your code works and mine doesnt: 
> ```dart
> theme: ThemeData.from(colorScheme: ColorScheme.light()).copyWith(
>   buttonTheme: ButtonThemeData(buttonColor: Colors.green),
> ),
> darkTheme: ThemeData.from(colorScheme: ColorScheme.dark()),
> ```

ColorSchemes are documented as still being a process and not everyone (especially those who already finalized their theming before ColorSchemes were published) knows to use them. That's essentially what my tests above established. That no matter what changes I made to the theme, only adding `ThemeData.buttonTheme.colorScheme` worked. Can this be fixed to be backwards-compatible with "normal" theming until ColorSchemes fully replace the parameters in the ThemeData constructor?
